swupd/delta.go

```diff
@@ -0,0 +1,160 @@
+package swupd
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/clearlinux/mixer-tools/helpers"
+	"github.com/pkg/errors"
+)
+
+// CreateDeltas creates all delta files between the previous and current
+// version of the supplied manifest, returning a list of files that
+// delta creation failed for.
+func CreateDeltas(manifest *Manifest, statedir string) ([]*File, error) {
+	var c config
+	c, err := getConfig(statedir)
+	if err != nil {
+		return nil, err
+	}
+
+	var oldManifest *Manifest
+	var newManifest *Manifest
+
+	if oldManifest, err = ParseManifestFile(fmt.Sprintf("%s/%d/%s", c.outputDir, manifest.Header.Previous, manifest.Name)); err != nil {
+		return nil, err
+	}
+	if newManifest, err = ParseManifestFile(fmt.Sprintf("%s/%d/%s", c.outputDir, manifest.Header.Version, manifest.Name)); err != nil {
```

Comment:
same - `filepath.Join()`

Reply:
woops, this was supposed to be changed before pushing 👍